### PR TITLE
fix: Duration Display picks the wrong sign and overflows at i64::MIN

### PR DESCRIPTION
**Pull request by chatman-media**
Display was reading the sign off the raw (unnormalized) seconds/nanos instead of the normalized value, so something like `Duration { seconds: 1, nanos: -1 }` — which normalizes to a positive 0.999999999s — printed as `-0.999999999s`. On top of that, both Display and `TryFrom<Duration> for time::Duration` call signed `.abs()`/negation on i64, which panics once seconds lands on i64::MIN (or silently misbehaves in release builds).

Swapped those to `unsigned_abs()` and made Display derive the sign from the normalized value. Added a couple of tests for the sign flip and the i64::MIN edge cases on both paths.

### CODE REVIEW

File: prost-types/src/duration.rs [hunk 1]
```diff
@@ -386,6 +386,60 @@ mod tests {
         );
     }
 
+    #[cfg(feature = "std")]
```
Comment (caspermeijn) [inline prost-types/src/duration.rs:389]: std is not required for display. Please remove feature requirement

**Review by caspermeijn — CHANGES_REQUESTED** [review]
Thank you for this fix including tests. The code looks good. I have a small nit pick.
